lib/kay/src/compact_dict.rs

```diff
@@ -32,21 +41,25 @@ impl <K: Eq + Copy, V: Compact + Clone, A: Allocator> CompactDict<K, V, A> {
         None
     }
 
+    /// Lookup the item in the dict, but also swaps the item with the first element so lookup for
+    /// that item will be faster
     pub fn get_mru(&mut self, query: K) -> Option<&V> {
         for i in 0..self.keys.len() {
             if self.keys[i] == query {
-                self.keys.swap(0, i);
+                self.keys.swap(0, i); // TODO: Is is better to swap with the one in front instead of the first?
```

Comment:
After thinking about it a little more, I'm pretty sure the current method is definitely suboptimal, because if you access more than one thing regularly, it will be exactly the same as the normal ordering, but with the added overhead of swapping.
E.g. a hashmap of len 5, and you access element 3, 4, 3, 4 ... in that order, it would find element 3, swap that to the front, then it would need to find element 4, and swap that ..., and doesn't promote them further forwards.